R/literal_coercion_linter.R

```diff
@@ -84,13 +84,18 @@ literal_coercion_linter <- function() {
 
     bad_expr <- xml2::xml_find_all(xml, xpath)
 
+    coercion_str <- xml2::xml_text(bad_expr)
+    # the linter logic should ensure that it's safe to run eval() here
+    literal_equivalent_str <- vapply(str2expression(coercion_str), function(expr) deparse1(eval(expr)), character(1L))
```

Comment:
Oh, hmm, for the `rlang::` coercions, this `eval()` will induce a dependency on {rlang}.

I see two options here:

 1. Attempt to translate `rlang::lgl()` into `as.logical()` and run that `eval()` instead
 2. Check `requireNamespace("rlang")`, using the original generic message if it's not available and `eval()` otherwise. Would require us (IINM) to add `rlang` to Suggests

The latter is easier to implement (in particular, inputs like `lgl(1,)` make things more complicated for option 1), and avoids the possibility that `as.logical()` and `lgl()` differ sometimes.

Any opinions?

Reply:
I'm fine with option 2. Option 1 seems like bugs waiting to happen.